### PR TITLE
Java dispatcher group affinity through searchPath

### PR DESCRIPTION
**Pull request by ollivir**
Also removed `Query` from `LoadBalancer` API since it's not used there.

### CODE REVIEW

File: container-search/src/main/java/com/yahoo/search/dispatch/LoadBalancer.java [hunk 1]
```diff
@@ -52,7 +52,7 @@ public LoadBalancer(SearchCluster searchCluster, boolean roundRobin) {
      * @param rejectedGroups if not null, the load balancer will only return groups with IDs not in the set
```
Comment (baldersheim) [inline container-search/src/main/java/com/yahoo/search/dispatch/LoadBalancer.java]: You need to remove the javadoc for query param to make the build pass.

**Review by baldersheim — CHANGES_REQUESTED** [review]
Apart from the java doc this looks fine.

**Review by bratseth — APPROVED** [review]
You might want to rename to just "takeGroup" now.

File: container-search/src/main/java/com/yahoo/search/dispatch/LoadBalancer.java (hunk 1 above)
Comment (bratseth) [reply container-search/src/main/java/com/yahoo/search/dispatch/LoadBalancer.java]: Done

**Pull request merged**